thrust/thrust/system/cuda/detail/uninitialized_copy.h

```diff
@@ -77,7 +80,22 @@ template <class Derived, class InputIt, class Size, class OutputIt>
 OutputIt _CCCL_HOST_DEVICE
 uninitialized_copy_n(execution_policy<Derived>& policy, InputIt first, Size count, OutputIt result)
 {
-  cuda_cub::parallel_for(policy, __uninitialized_copy::functor<InputIt, OutputIt>{first, result}, count);
+  // if the output type is trivially constructible from the input, it has no side effect and we can skip placement new
+  // and calling a constructor. If the type is also trivially relocatable, we can construct the instances in a kernel
+  // and memcpy to the destination. In that case, transform offers several fast paths.
+  using ctor_arg_t = thrust::detail::raw_reference_t<::cuda::std::iter_reference_t<InputIt>>;
+  using output_t   = thrust::detail::it_value_t<OutputIt>;
+  if constexpr (::cuda::std::is_trivially_constructible_v<output_t, ctor_arg_t>
+                && thrust::is_trivially_relocatable_v<output_t>)
+  {
+    // TODO(bgruber): converting it+count to it+it may be wasteful, since transform internally converts back
+    cuda_cub::transform(policy, first, first + count, result, ::cuda::std::identity{});
```

Comment:
We should probably implement `thrust::transform_n` in the generic system and offer a dedicated overload in the CUDA system, and then map every algorithm using `transform` to `transform_n`.